Source/Particles/WarpXParticleContainer.H

```diff
@@ -332,7 +332,7 @@ public:
     /**
      * \brief Virtual method to resample the species. Overriden by PhysicalParticleContainer.
      */
-    virtual void resample (const Resampling& /*resampler*/, const int /*timestep*/) {};
+    virtual void resample (const Resampling& /*resampler*/, const int /*timestep*/) = 0;
```

Comment:
revert (to consider in a follow-up)
```suggestion
    virtual void resample (const Resampling& /*resampler*/, const int /*timestep*/) {}
```